In [ ]:
import pandas as pd

In [ ]:
import numpy as np

In [ ]:
data = pd.read_csv('iris.csv')
data.head(10)

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa
5           5.4          3.9           1.7          0.4  setosa
6           4.6          3.4           1.4          0.3  setosa
7           5.0          3.4           1.5          0.2  setosa
8           4.4          2.9           1.4          0.2  setosa
9           4.9          3.1           1.5          0.1  setosa

In [ ]:
for i in data.index:
  if data.loc[i, 'sepal_length'] >=8.0:
    data.loc[i, 'sepal_length'] = 7.0
print(data.to_string())  

     sepal_length  sepal_width  petal_length  petal_width     species
0             5.1          3.5           1.4          0.2      setosa
1             4.9          3.0           1.4          0.2      setosa
2             4.7          3.2           1.3          0.2      setosa
3             4.6          3.1           1.5          0.2      setosa
4             5.0          3.6           1.4          0.2      setosa
5             5.4          3.9           1.7          0.4      setosa
6             4.6          3.4           1.4          0.3      setosa
7             5.0          3.4           1.5          0.2      setosa
8             4.4          2.9           1.4          0.2      setosa
9             4.9          3.1           1.5          0.1      setosa
10            5.4          3.7           1.5          0.2      setosa
11            4.8          3.4           1.6          0.2      setosa
12            4.8          3.0           1.4          0.1      setosa
13            4.3   

In [ ]:
print(data['sepal_length'])

0      5.1
1      4.9
2      4.7
3      4.6
4      5.0
      ... 
145    6.7
146    6.3
147    6.5
148    7.0
149    5.9
Name: sepal_length, Length: 150, dtype: float64


In [ ]:
data.loc[0, 'sepal_length'] = np.nan
data

sepal_length  sepal_width  petal_length  petal_width    species
0             NaN          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           7.0          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

In [ ]:
for i in data.index:
  if pd.isnull(data.loc[i,'sepal_length']) == True:
    data.drop(data.index[i], inplace=True) 
data

sepal_length  sepal_width  petal_length  petal_width    species
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
5             5.4          3.9           1.7          0.4     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           7.0          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[149 rows x 5 columns]

In [ ]:
data.head(40)

sepal_length  sepal_width  petal_length  petal_width species
1            4.9          3.0           1.4          0.2  setosa
2            4.7          3.2           1.3          0.2  setosa
3            4.6          3.1           1.5          0.2  setosa
4            5.0          3.6           1.4          0.2  setosa
5            5.4          3.9           1.7          0.4  setosa
6            4.6          3.4           1.4          0.3  setosa
7            5.0          3.4           1.5          0.2  setosa
8            4.4          2.9           1.4          0.2  setosa
9            4.9          3.1           1.5          0.1  setosa
10           5.4          3.7           1.5          0.2  setosa
11           4.8          3.4           1.6          0.2  setosa
12           4.8          3.0           1.4          0.1  setosa
13           4.3          3.0           1.1          0.1  setosa
14           5.8          4.0           1.2          0.2  setosa
15           5.7          4.4           1.5          0.4  setosa
16           5.4          3.9           1.3          0.4  setosa
17           5.1          3.5           1.4          0.3  setosa
18           5.7          3.8           1.7          0.3  setosa
19           5.1          3.8           1.5          0.3  setosa
20           5.4          3.4           1.7          0.2  setosa
21           5.1          3.7           1.5          0.4  setosa
22           4.6          3.6           1.0          0.2  setosa
23           5.1          3.3           1.7          0.5  setosa
24           4.8          3.4           1.9          0.2  setosa
25           5.0          3.0           1.6          0.2  setosa
26           5.0          3.4           1.6          0.4  setosa
27           5.2          3.5           1.5          0.2  setosa
28           5.2          3.4           1.4          0.2  setosa
29           4.7          3.2           1.6          0.2  setosa
30           7.0          3.1           1.6          0.2  setosa
31           5.4          3.4           1.5          0.4  setosa
32           5.2          4.1           1.5          0.1  setosa
33           5.5          4.2           1.4          0.2  setosa
34           4.9          3.1           1.5          0.1  setosa
35           5.0          3.2           1.2          0.2  setosa
36           5.5          3.5           1.3          0.2  setosa
37           4.9          3.1           1.5          0.1  setosa
38           4.4          3.0           1.3          0.2  setosa
39           5.1          3.4           1.5          0.2  setosa
40           5.0          3.5           1.3          0.3  setosa

In [ ]:
print(data.duplicated().to_string())

1      False
2      False
3      False
4      False
5      False
6      False
7      False
8      False
9      False
10     False
11     False
12     False
13     False
14     False
15     False
16     False
17     False
18     False
19     False
20     False
21     False
22     False
23     False
24     False
25     False
26     False
27     False
28     False
29     False
30     False
31     False
32     False
33     False
34      True
35     False
36     False
37      True
38     False
39     False
40     False
41     False
42     False
43     False
44     False
45     False
46     False
47     False
48     False
49     False
50     False
51     False
52     False
53     False
54     False
55     False
56     False
57     False
58     False
59     False
60     False
61     False
62     False
63     False
64     False
65     False
66     False
67     False
68     False
69     False
70     False
71     False
72     False
73     False
74     False
75     False
76     False
77     False

In [ ]:
data.head(40)

sepal_length  sepal_width  petal_length  petal_width species
1            4.9          3.0           1.4          0.2  setosa
2            4.7          3.2           1.3          0.2  setosa
3            4.6          3.1           1.5          0.2  setosa
4            5.0          3.6           1.4          0.2  setosa
5            5.4          3.9           1.7          0.4  setosa
6            4.6          3.4           1.4          0.3  setosa
7            5.0          3.4           1.5          0.2  setosa
8            4.4          2.9           1.4          0.2  setosa
9            4.9          3.1           1.5          0.1  setosa
10           5.4          3.7           1.5          0.2  setosa
11           4.8          3.4           1.6          0.2  setosa
12           4.8          3.0           1.4          0.1  setosa
13           4.3          3.0           1.1          0.1  setosa
14           5.8          4.0           1.2          0.2  setosa
15           5.7          4.4           1.5          0.4  setosa
16           5.4          3.9           1.3          0.4  setosa
17           5.1          3.5           1.4          0.3  setosa
18           5.7          3.8           1.7          0.3  setosa
19           5.1          3.8           1.5          0.3  setosa
20           5.4          3.4           1.7          0.2  setosa
21           5.1          3.7           1.5          0.4  setosa
22           4.6          3.6           1.0          0.2  setosa
23           5.1          3.3           1.7          0.5  setosa
24           4.8          3.4           1.9          0.2  setosa
25           5.0          3.0           1.6          0.2  setosa
26           5.0          3.4           1.6          0.4  setosa
27           5.2          3.5           1.5          0.2  setosa
28           5.2          3.4           1.4          0.2  setosa
29           4.7          3.2           1.6          0.2  setosa
30           7.0          3.1           1.6          0.2  setosa
31           5.4          3.4           1.5          0.4  setosa
32           5.2          4.1           1.5          0.1  setosa
33           5.5          4.2           1.4          0.2  setosa
34           4.9          3.1           1.5          0.1  setosa
35           5.0          3.2           1.2          0.2  setosa
36           5.5          3.5           1.3          0.2  setosa
37           4.9          3.1           1.5          0.1  setosa
38           4.4          3.0           1.3          0.2  setosa
39           5.1          3.4           1.5          0.2  setosa
40           5.0          3.5           1.3          0.3  setosa

In [ ]:
print(data.corr())

              sepal_length  sepal_width  petal_length  petal_width
sepal_length      1.000000    -0.095861      0.849434     0.797304
sepal_width      -0.095861     1.000000     -0.415218    -0.350733
petal_length      0.849434    -0.415218      1.000000     0.962314
petal_width       0.797304    -0.350733      0.962314     1.000000


In [ ]:
data

sepal_length  sepal_width  petal_length  petal_width    species
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
5             5.4          3.9           1.7          0.4     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           7.0          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[149 rows x 5 columns]

In [ ]:
diab = pd.read_csv('diabetes.csv')
diab.columns=["Pregnancies", "Glucose", "BloodPressure", "SkinThickness(mm)", "Insulin(mu U/ml)", "BMI", "DiabetesPedigreeFunction", "Age(Years)", "Outcome"]
print(diab)

     Pregnancies  Glucose  ...  Age(Years)  Outcome
0              1       85  ...          31        0
1              8      183  ...          32        1
2              1       89  ...          21        0
3              0      137  ...          33        1
4              5      116  ...          30        0
..           ...      ...  ...         ...      ...
762           10      101  ...          63        0
763            2      122  ...          27        0
764            5      121  ...          30       \N
765            1      126  ...          47       \N
766            1       93  ...          23       \N

[767 rows x 9 columns]


In [ ]:
dib = diab.loc[0:50,['Glucose', 'BloodPressure','BMI', 'Age(Years)']]
print(dib)

    Glucose  BloodPressure   BMI  Age(Years)
0        85             66  26.6          31
1       183             64  23.3          32
2        89             66  28.1          21
3       137             40  43.1          33
4       116             74  25.6          30
5        78             50  31.0          26
6       115              0  35.3          29
7       197             70  30.5          53
8       125             96   0.0          54
9       110             92  37.6          30
10      168             74  38.0          34
11      139             80  27.1          57
12      189             60  30.1          59
13      166             72  25.8          51
14      100              0  30.0          32
15      118             84  45.8          31
16      107             74  29.6          31
17      103             30  43.3          33
18      115             70  34.6          32
19      126             88  39.3          27
20       99             84  35.4          50
21      19

In [ ]:
dib.duplicated()

0     False
1     False
2     False
3     False
4     False
5     False
6     False
7     False
8     False
9     False
10    False
11    False
12    False
13    False
14    False
15    False
16    False
17    False
18    False
19    False
20    False
21    False
22    False
23    False
24    False
25    False
26    False
27    False
28    False
29    False
30    False
31    False
32    False
33    False
34    False
35    False
36    False
37    False
38    False
39    False
40    False
41    False
42    False
43    False
44    False
45    False
46    False
47    False
48    False
49    False
50    False
dtype: bool

In [ ]:
dib.dropna(inplace=True)
dib

Glucose  BloodPressure   BMI  Age(Years)
0        85             66  26.6          31
1       183             64  23.3          32
2        89             66  28.1          21
3       137             40  43.1          33
4       116             74  25.6          30
5        78             50  31.0          26
6       115              0  35.3          29
7       197             70  30.5          53
8       125             96   0.0          54
9       110             92  37.6          30
10      168             74  38.0          34
11      139             80  27.1          57
12      189             60  30.1          59
13      166             72  25.8          51
14      100              0  30.0          32
15      118             84  45.8          31
16      107             74  29.6          31
17      103             30  43.3          33
18      115             70  34.6          32
19      126             88  39.3          27
20       99             84  35.4          50
21      196             90  39.8          41
22      119             80  29.0          29
23      143             94  36.6          51
24      125             70  31.1          41
25      147             76  39.4          43
26       97             66  23.2          22
27      145             82  22.2          57
28      117             92  34.1          38
29      109             75  36.0          60
30      158             76  31.6          28
31       88             58  24.8          22
32       92             92  19.9          28
33      122             78  27.6          45
34      103             60  24.0          33
35      138             76  33.2          35
36      102             76  32.9          46
37       90             68  38.2          27
38      111             72  37.1          56
39      180             64  34.0          26
40      133             84  40.2          37
41      106             92  22.7          48
42      171            110  45.4          54
43      159             64  27.4          40
44      180             66  42.0          25
45      146             56  29.7          29
46       71             70  28.0          22
47      103             66  39.1          31
48      105              0   0.0          24
49      103             80  19.4          22
50      101             50  24.2          26

In [ ]:
dup = dib.duplicated()
for i in dup.index:
  if dup.loc[i]==True:
    print(i)

In [ ]:
# dib['Glucose'].mean()
dib['Glucose'].mode()
dib['Glucose'].median()
dib.corr()

Glucose  BloodPressure       BMI  Age(Years)
Glucose        1.000000       0.194435  0.212931    0.388261
BloodPressure  0.194435       1.000000  0.155112    0.362528
BMI            0.212931       0.155112  1.000000    0.045441
Age(Years)     0.388261       0.362528  0.045441    1.000000

**Reference : Diabetes data of India from UCI repository**


1. Total rows = 767
2. Rows taken = 50
3. Columns of interest taken = 4 out of 9
4. Columns = ['Glucose', 'BloodPressure', 'BMI', 'Age']
5. Null values removed
6. Duplicated values = None
7. Glucose level mean of sample taken = 125.980
8. Glucose level mode = 103
9. Glucose level median = 117.0
10. Correlation: The values of Glucose are more related to Age, then BMI and then Blood pressure of patients.











